File: BackEnd/src/main/java/com/codeclan/example/PetApp/models/Booking.java
package com.codeclan.example.PetApp.models;

import com.fasterxml.jackson.annotation.JsonIgnore;

import javax.persistence.*;

@Entity
@Table(name = "bookings")


public class Booking {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @JsonIgnore
    @ManyToOne
    @JoinColumn(name = "pet_id", nullable = false)
    private Pet pet;

    @JsonIgnore
    @ManyToOne
    @JoinColumn(name = "service_id", nullable = false)
    private Service service;

    @Column (name = "date")
    private String date;


    public Booking (String date, Pet pet, Service service) {

        this.pet = pet;
        this.service = service;
        this.date = date;

    }

    public Booking() {
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Pet getPet() {
        return pet;
    }

    public void setPet(Pet pet) {
        this.pet = pet;
    }

    public Service getService() {
        return service;
    }

    public void setService(Service service) {
        this.service = service;
    }

    public String getDate() {
        return date;
    }

    public void setDate(String date) {
        this.date = date;
    }
}

File: frontend/src/components/owners/Booking.js
import React from 'react';
import {Link} from 'react-router-dom';

const Booking = (props) => {

  if (!props.booking) {
    return null;
  }

  return(
    <React.Fragment>
      <p>ID: {props.booking.id}</p>
      <p>Date: {props.booking.date}</p>
      <p>Pet Name: {props.booking.pet.name}</p>
      <p>Pet Type: {props.booking.pet.type}</p>
    </React.Fragment>

  )
}

export default Booking;
